core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp

```diff
@@ -1077,11 +1216,18 @@ imm_set_value(Imm &imm, void *data, uint32 bytes)
  */
 static bool
 st_imm_to_base_imm_offset_imm(x86::Assembler &a, uint32 bytes_dst,
-                              void *data_src, int32 base, int32 offset)
+                              void *data_src, int32 base, int32 offset,
+                              bool atomic)
 {
     x86::Mem m((uintptr_t)(base + offset), bytes_dst);
     Imm imm;
     imm_set_value(imm, data_src, bytes_dst);
+#if WASM_ENABLE_SHARED_MEMORY != 0
+    uint32 reg_no_src = mov_imm_to_free_reg(a, imm, bytes_dst);
+    if (atomic) {
+        return xchg_r_to_m(a, bytes_dst, JIT_REG_KIND_I64, m, reg_no_src);
```

Comment:
Is it right to use `JIT_REG_KIND_I64` here?

Reply:
In this context, this parameter is only used for assert inside `xchg_r_to_m`, which does not influence the correctness. No matter the origin type of store value(I32 or I64),  the correctness only depends on how many bytes it actually stores since the store doesn't have any return value.